ProtectionScan/Features/MainFeature.cs

```diff
@@ -199,5 +216,52 @@ private void WriteProtectionResultFile(string path, Dictionary<string, List<stri
             // Dispose of the writer
             sw?.Dispose();
         }
+        
+#if NETCOREAPP
+        /// <summary>
+        /// Write the protection results from a single path to a json file, if possible
+        /// </summary>
+        /// <param name="path">File or directory path</param>
+        /// <param name="protections">Dictionary of protections found, if any</param>
+        private static void WriteProtectionResultJson(string path, Dictionary<string, List<string>> protections)
+        {
+            if (protections == null)
+            {
+                Console.WriteLine($"No protections found for {path}");
+                return;
+            }
+
+            // Attempt to open a protection file for writing
+            StreamWriter? jsw = null;
+            try
+            {
+                jsw = new StreamWriter(File.OpenWrite($"protection-{DateTime.Now:yyyy-MM-dd_HHmmss.ffff}.json"));
+            }
+            catch { }
+
+            // Create the output data
+            string serializedData = System.Text.Json.JsonSerializer.Serialize(protections, JsonSerializerOptions);
+
+            // Write the output data
+            // TODO: this prints plus symbols wrong, probably some other things too
+            jsw?.WriteLine(serializedData);
+            jsw?.Flush();
+            
+            // Dispose of the writer
+            jsw?.Dispose();
```

Comment:
If this try/catch fails, then the rest of the method does nothing. It is probably easier to put all of the logic into the try/catch and have it catch the error external to everything.